mungegithub/mungers/check-labels.go

```diff
@@ -0,0 +1,165 @@
+}
+
+func init() {
+	RegisterMungerOrDie(&CheckLabelsMunger{})
+}
+
+// Name is the name usable in --pr-mungers
+func (c *CheckLabelsMunger) Name() string { return "check-labels" }
+
+// RequiredFeatures is a slice of 'features' that must be provided.
+func (c *CheckLabelsMunger) RequiredFeatures() []string { return []string{features.RepoFeatureName} }
+
+// Initialize will initialize the munger.
+func (c *CheckLabelsMunger) Initialize(config *githubhelper.Config, features *features.Features) error {
+	if len(c.labelFilePath) == 0 {
+		glog.Fatalf("No --label-file= supplied, cannot check labels")
+	}
+	c.labelAccessor = config
+	c.features = features
+	c.reader = func() (string, error) {
+		bytes, err := ioutil.ReadFile(c.labelFilePath)
+		if err != nil {
+			return "", fmt.Errorf("Could not read label file: %v", err)
+		}
+		return string(bytes), nil
+	}
+
+	if _, err := os.Stat(c.labelFilePath); os.IsNotExist(err) {
+		return fmt.Errorf("Failed to stat the check label config: %v", err)
+	}
+
+	return nil
+}
+
+func (c *CheckLabelsMunger) getSha(fileContents string) string {
+	h := sha1.New()
+	h.Write([]byte(fileContents))
+	bs := h.Sum(nil)
+	return string(bs)
+}
+
+// EachLoop is called at the start of every munge loop
+func (c *CheckLabelsMunger) EachLoop() error {
+	fileContents, err := c.reader()
+	if err != nil {
+		glog.Errorf("Failed to read the check label config: %v", err)
+		return err
+	}
+	sha := c.getSha(fileContents)
+	if c.prevSha != sha {
+		// Get all labels from file.
+		fileLabels := map[string][]*github.Label{}
+		if err := yaml.NewYAMLToJSONDecoder(strings.NewReader(fileContents)).Decode(&fileLabels); err != nil {
+			glog.Errorf("Failed to decode the check label config: %v", err)
+			return err
+		}
+
+		// Get all labels from repository.
+		repoLabels, err := c.labelAccessor.GetLabels()
+		if err != nil {
+			return err
+		}
+		c.createNewLabels(repoLabels, fileLabels["labels"])
+		c.prevSha = sha
+	}
+	return nil
+}
+
+func (c *CheckLabelsMunger) createNewLabels(repoLabels, fileLabels []*github.Label) {
+	repoLabelSet := sets.NewString()
+	for _, repoLabel := range repoLabels {
+		repoLabelSet.Insert(*repoLabel.Name)
+	}
+
+	// Compare against labels in local file.
+	for _, label := range fileLabels {
+		if !repoLabelSet.Has(*label.Name) {
+			err := c.labelAccessor.AddLabel(label)
+			if err != nil {
+				glog.Errorf("Error %s in adding label %s", err, *label.Name)
+			}
+		}
+	}
+}
+
+// AddFlags will add any request flags to the cobra `cmd`.
+func (c *CheckLabelsMunger) AddFlags(cmd *cobra.Command, config *githubhelper.Config) {
+	cmd.Flags().StringVar(&c.labelFilePath, "label-file", "", "Path from repository root to file containing"+
+		" list of labels")
+}
+
+// Munge is unused by this munger.
+func (c *CheckLabelsMunger) Munge(obj *githubhelper.MungeObject) {}
+
+// This function can be used to write a check-labels config file.
+func writeLabelFile(c *CheckLabelsMunger, filename string) error {
```

Comment:
Nothing calls it, but it is the way to write out valid labels-config yaml files when building the list from an existing repository. I thought it might be useful to include it.